### ISSUE TITLE
Cities to convert

### ISSUE THREAD

**Issue by meetar** [polygon-city/polygon-city-cli#22]
based on https://github.com/polygon-city/polygon-city-app/issues/48

**Comment by robhawkes** [comment]
I'll keep this table updated with data sources…

| City | Licence | Format | Notes |
| :-- | :-- | :-- | :-- |
| [Berlin, Germany](http://www.businesslocationcenter.de/berlin3d-downloadportal/documents/manual.en.html) | [Licence](http://www.businesslocationcenter.de/berlin3d-downloadportal/documents/terms.en.html) | CityGML (LOD2) | Few, _huge_ CityGML files |
| [Toronto, Canada](http://www1.toronto.ca/wps/portal/contentonly?vgnextoid=d431d477f9a3a410VgnVCM10000071d60f89RCRD&vgnextchannel=1a66e03bb8d1e310VgnVCM10000071d60f89RCRD) | [Licence](http://www1.toronto.ca/wps/portal/contentonly?vgnextoid=4a37e03bb8d1e310VgnVCM10000071d60f89RCRD) | DWG, DGN & SKP |  |
| [Lyon, France](http://data.grandlyon.com/search/?Q=citygml) | [Licence](https://download.data.grandlyon.com/files/grandlyon/LicenceOuverte.pdf) | CityGML (LOD2) | Models are at block level |
| [Hamburg, Germany](http://suche.transparenz.hamburg.de/dataset/3d-stadtmodell-hamburg) | [Licence](https://www.govdata.de/dl-de/by-2-0) | CityGML (LOD1) | LOD2 due late 2016 |
| [Montreal, Canada](http://donnees.ville.montreal.qc.ca/dataset/maquette-numerique-batiments-citygml-lod2-avec-textures) | [Licence](http://creativecommons.org/licenses/by/4.0/) | CityGML (LOD2) & 3DM |  |
| [Rotterdam, Netherlands](http://rotterdamopendata.nl/dataset/rotterdam-3d-bestanden) | Undisclosed | CityGML (LOD2) | Validation issues during conversion |
| [NYC, US](http://www.gis.bgu.tum.de/en/projects/new-york-city-3d/) | Undisclosed | CityGML (LOD1) |  |
| [NYC, US Official!](https://data.cityofnewyork.us/City-Government/3-D-Building-Model/tnru-abg2) | ? | CityGML (LOD1/LOD2) | A few large files |
| [Linz, Austria](http://geo.data.linz.gv.at/katalog/geodata/3d_geo_daten_lod2/) | Undisclosed | CityGML (LOD2) | Lots of small CityGML files |
| [Copenhagen, Denmark](http://data.kk.dk/dataset/bymodel-3d-kobenhavn) | [Licence](https://digitaliser.dk/pages/ResourceView.aspx?ResourceView=2432531) | DWG, DGN & DXF |  |

**Comment by robhawkes** [comment]
I had a play with the new, official NYC data and the results are promising!

![](http://cl.ly/2q3h2c1D3Z0J/Screen%20Shot%202016-04-20%20at%2011.12.02.png)

Unfortunately, it seems like the converter isn't working correctly for some buildings (eg. the library) but I've not put any effort in to work out what exactly is failing here but the result is that same faces are flipped. Looks like either the geometry is invalid, or the obj-conversion / repair operations are doing things they shouldn't be (like flipping faces that were actually valid to begin with). It may actually make sense to provide an option to disable validation checks and repairs as if the data is 100% valid anyway then it may produce more reliable results.

![](http://cl.ly/3O432r3E2y23/Screen%20Shot%202016-04-20%20at%2011.20.10.png)

**Comment by meetar** [comment]
Yes, I noticed that too but haven't had the chance to look more closely at it. I like the idea of making validation an option, though.

**Comment by robhawkes** [comment]
From what I can see there are a few buildings that have overlapping geometry too which causes a lot of z-fighting, so my money is on a combination of bad / invalid geometry and edge-case issues with the converter / repair system I wrote.

Probably best to run it through [val3dity](https://github.com/tudelft3d/val3dity) and see what comes back.
